airflow/contrib/operators/bigquery_operator.py

```diff
@@ -569,3 +568,65 @@ def execute(self, context):
             project_id=self.project_id,
             dataset_id=self.dataset_id
         )
+
+
+class BigQueryCreateEmptyDatasetOperator(BaseOperator):
+    """"
+    This operator is used to create new dataset for your Project in Big query.
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets#resource
+
+    :param project_id: The name of the project where we want to create the dataset.
+           Not need to provide, if projectId in dataset_resource.
+    :type project_id: str
+    :param dataset_id: The id of dataset. Not need to provide,
+           if datasetId in dataset_resource.
+    :type dataset_id: str
+    :param dataset_resource: Dataset resource that must be provided
+           with request body. More info:
+           https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets#resource
+    :type dataset_resource: dict
+
+        **Example**: ::
+
+            create_new_dataset = BigQueryCreateEmptyDatasetOperator(
+                                    dataset_id = 'new-dataset',
+                                    project_id = 'my-project',
+                                    dataset_resource = {"friendlyName": "New Dataset"}
+                                    bigquery_conn_id='_my_gcp_conn_',
+                                    task_id='newDatasetCreator',
+                                    dag=dag)
+    """
```

Comment:
Can you also add one more example where you specify different values for `dataset_id` param and datasetID in `dataset_resource` and a line saying which value would be kept and which would be overrriden?

Reply:
@kaxil , I don't override. As we discussed in https://github.com/apache/incubator-airflow/pull/3733 in case if in some dict with params exist same param what in args and it's provided by user in 2 places: I raise error with information about duplicates and call for user to remove one of them.